src/client/sandbox/cookie/server-sync.js

```diff
@@ -0,0 +1,122 @@
+import nativeMethods from '../../sandbox/native-methods';
+import XhrSandbox from '../../sandbox/xhr';
+import settings from '../../settings';
+import { stringify as stringifyJSON } from '../../json';
+import { isIE9 } from '../../utils/browser';
+
+const SWITCH_BACK_TO_ASYNC_XHR_DELAY = 2000;
+
+export default class ServerSync {
+    constructor () {
+        this.useAsyncXhr = true;
+        this.queue       = [];
+        this.sendedQueue = null;
+        this.activeReq   = null;
+
+        // NOTE: When unloading, we should switch to synchronous XHR to be sure that we won't lose any cookies.
+        nativeMethods.windowAddEventListener.call(window, 'beforeunload', () => this._beforeUnloadHandler(), true);
+    }
+
+    static _createXMLHttpRequest (isAsync) {
+        var xhr = XhrSandbox.createNativeXHR();
+
+        xhr.open('POST', settings.get().cookieSyncUrl, isAsync);
```

Comment:
Rename parameter to `async`.
It called as `async` in [documentation](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/open)

Why do we need separate url for cookie synchronization?
I prefer to have an additional check in `Proxy._onServiceMessage` instead of adding the new service url.